Source/CMake/Properties.cmake

```diff
@@ -54,7 +55,7 @@ if(MSVC)
 
 	# Compiler
 	set(BS_COMPILER_FLAGS_COMMON "/GS- /W3 /GR- /WX- /MP /nologo /bigobj /wd\"4577\" /wd\"4530\"")
-	set(CMAKE_CXX_FLAGS "/DWIN32 /D_WINDOWS")
+	add_definitions(-DWIN32 -D_WINDOWS)
```

Comment:
This line appears to be causing an internal compiler error MSVC, along with other weirdness. Not sure why. I'll switch it back and merge without this change.

Reply:
That's amazing in a way, that different way of adding defines crashes msvc :D

Reply:
My thoughts exactly :/